clang/test/C/C2y/n3369.c

```diff
@@ -144,3 +177,13 @@ void test_compound_literals() {
   static_assert(_Countof((int[2]){}) == 2);
   static_assert(_Countof((int[]){1, 2, 3, 4}) == 4);	
 }
+
+static int test_f1();
+static int test_f2();  // expected-warning {{never defined}}
+
+void test_symbols() {
+  int a[global_num][global_num];
+
+  static_assert(_Countof(global_multi_array[test_f1()]) == 34);
+  (void)_Countof(a[test_f2()]);
```

Comment:
What is this one intended to test?

Reply:
That the symbol `test_f1` is not used at all.


We don't get a diagnostic due to having a declaration without a definition.  (But we get it in line 188 for `test_f2` because that's not an ICE.

Reply:
Sounds good! Worth adding a comment explaining that so others don't wonder as well.

Reply:
Done.  Feel free to reopen if the comment is unclear.